hibernate-core/src/main/java/org/hibernate/tool/schema/internal/SchemaTruncatorImpl.java

```diff
@@ -287,119 +270,9 @@ private static void checkExportIdentifier(Exportable exportable, Set<String> exp
 		exportIdentifiers.add( exportIdentifier );
 	}
 
-	//Woooooo, massive copy/paste from SchemaCreatorImpl!
-
-	private void applyImportSources(
-			ExecutionOptions options,
-			SqlScriptCommandExtractor commandExtractor,
-			boolean format,
-			Dialect dialect,
-			GenerationTarget... targets) {
-		final ServiceRegistry serviceRegistry = tool.getServiceRegistry();
-		final ClassLoaderService classLoaderService = serviceRegistry.getService( ClassLoaderService.class );
-
-		// I have had problems applying the formatter to these imported statements.
-		// and legacy SchemaExport did not format them, so doing same here
-		//final Formatter formatter = format ? DDLFormatterImpl.INSTANCE : FormatStyle.NONE.getFormatter();
-		final Formatter formatter = FormatStyle.NONE.getFormatter();
-
-		Object importScriptSetting = options.getConfigurationValues().get( HBM2DDL_LOAD_SCRIPT_SOURCE );
-		if ( importScriptSetting == null ) {
-			importScriptSetting = options.getConfigurationValues().get( JAKARTA_HBM2DDL_LOAD_SCRIPT_SOURCE );
-		}
-		String charsetName = (String) options.getConfigurationValues().get( HBM2DDL_CHARSET_NAME );
-
-		if ( importScriptSetting != null ) {
-			final ScriptSourceInput importScriptInput = interpretScriptSourceSetting( importScriptSetting, classLoaderService, charsetName );
-			applyScript( options, commandExtractor, dialect, importScriptInput, formatter, targets );
-		}
-
-		final String importFiles = ConfigurationHelper.getString(
-				AvailableSettings.HBM2DDL_IMPORT_FILES,
-				options.getConfigurationValues(),
-				SchemaCreatorImpl.DEFAULT_IMPORT_FILE
-		);
-
-		for ( String currentFile : StringHelper.split( ",", importFiles ) ) {
-			final String resourceName = currentFile.trim();
-			if ( resourceName.isEmpty() ) {
-				//skip empty resource names
-				continue;
-			}
-			final ScriptSourceInput importScriptInput = interpretLegacyImportScriptSetting( resourceName, classLoaderService, charsetName );
-			applyScript( options, commandExtractor, dialect, importScriptInput, formatter, targets );
-		}
-	}
-
-	private ScriptSourceInput interpretLegacyImportScriptSetting(
-			String resourceName,
-			ClassLoaderService classLoaderService,
-			String charsetName) {
-		try {
-			final URL resourceUrl = classLoaderService.locateResource( resourceName );
-			if ( resourceUrl == null ) {
-				return ScriptSourceInputNonExistentImpl.INSTANCE;
-			}
-			else {
-				return new ScriptSourceInputFromUrl( resourceUrl, charsetName );
-			}
-		}
-		catch (Exception e) {
-			throw new SchemaManagementException( "Error resolving legacy import resource : " + resourceName, e );
-		}
-	}
-
-	/**
-	 * Intended for use from tests
-	 */
-	@Internal
-	public void doTruncate(
-			Metadata metadata,
-			final boolean manageNamespaces,
-			GenerationTarget... targets) {
-		final ServiceRegistry serviceRegistry =
```

Comment:
sooo... these were used in Search tests 🥲, not anymore, I guess ...

Reply:
ohh, no... not testing this method or truncator in general, but using it to clean the test data 🙈

Reply:
I mean we're talking about an unused method annotated `@Internal` of an `Impl` class in an `internal` package.

We really need to avoid cross-project dependencies going through methods like that. That's why we define SPIs.

Reply:
>  using it to clean the test data

The you should be using the API for that, not depending on internal implementation details.

If that API doesn't work, then you should propose introduction of an SPI.

Otherwise, when we're working on ORM, we have _absolutely no warning_ that we've done something that might break Search, and indeed, _Search deserves to break_.

Reply:
Do you happen to use the `@SessionFactory` stuff for testing @marko-bekhta ?  If so I added `SessionFactoryScope#dropData`.  Otherwise you could just use `			`sessionFactory.getSchemaManager().truncateMappedObjects()`

Reply:
ohh Search has a set of its own extensions to deal with tests 🙈  so we don't leverage the `@SessionFactory` (at least not yet 😉).

> Otherwise you could just use sessionFactory.getSchemaManager().truncateMappedObjects()

right ... that's what we do, it's just that we have this extension to deal with a multitenancy, and it has a custom schema manager with overrides, that's where that method was used from 😃.